src/CaptureServiceBase/CloudCollectorStartStopCaptureRequestWaiter.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "CaptureServiceBase/CloudCollectorStartStopCaptureRequestWaiter.h"
+
+#include "OrbitBase/Logging.h"
+
+using orbit_grpc_protos::CaptureOptions;
+
+namespace orbit_capture_service_base {
+
+CaptureOptions CloudCollectorStartStopCaptureRequestWaiter::WaitForStartCaptureRequest() {
+  // This is blocked until StartCapture is called.
+  while (!start_requested_) {
+  }
```

Comment:
I'd recommend to use an `absl::Mutex` and a simple `bool` instead of `atomic<bool>` because this allows us to use `mutex.Await` here instead of a busy loop.

Reply:
Done! Thanks for the suggestion!